app/screens/post_priority_picker/post_priority_picker.tsx

```diff
@@ -0,0 +1,258 @@
+    isPersistenNotificationsEnabled: boolean;
+    postPriority: PostPriority;
+    updatePostPriority: (data: PostPriority) => void;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        backgroundColor: theme.centerChannelBg,
+    },
+    titleContainer: {
+        alignItems: 'center',
+        flexDirection: 'row',
+    },
+    title: {
+        color: theme.centerChannelColor,
+        ...typography('Heading', 600, 'SemiBold'),
+    },
+    betaContainer: {
+        backgroundColor: PostPriorityColors.IMPORTANT,
+        borderRadius: 4,
+        paddingHorizontal: 4,
+        marginLeft: 8,
+    },
+    beta: {
+        color: '#fff',
+        ...typography('Body', 25, 'SemiBold'),
+    },
+
+    optionsContainer: {
+        paddingTop: 12,
+    },
+
+    optionsSeparator: {
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        height: 1,
+    },
+
+    toggleOptionContainer: {
+        marginTop: 16,
+    },
+}));
+
+const PostPriorityPicker = ({
+    componentId, isPostAcknowledgementEnabled, isPersistenNotificationsEnabled,
+    postPriority, updatePostPriority,
+}: Props) => {
+    const {bottom} = useSafeAreaInsets();
+    const intl = useIntl();
+    const isTablet = useIsTablet();
+    const theme = useTheme();
+
+    const style = getStyleSheet(theme);
+
+    const close = useCallback(() => {
+        return dismissBottomSheet(Screens.POST_PRIORITY_PICKER);
+    }, []);
+
+    useNavButtonPressed(POST_PRIORITY_PICKER_BUTTON, componentId, close, []);
+
+    const [data, setData] = useState<PostPriority>(postPriority);
+
+    const displayPersistentNotifications = isPersistenNotificationsEnabled && data.priority === PostPriorityType.URGENT;
+
+    const snapPoints = useMemo(() => {
+        let COMPONENT_HEIGHT = 280;
+
+        if (isPostAcknowledgementEnabled) {
+            COMPONENT_HEIGHT += 75;
+        }
+
+        if (displayPersistentNotifications) {
+            COMPONENT_HEIGHT += 75;
+        }
+
+        return [1, bottomSheetSnapPoint(1, COMPONENT_HEIGHT, bottom)];
+    }, [displayPersistentNotifications, isPostAcknowledgementEnabled]);
+
+    const handleUpdatePriority = useCallback((priority: PostPriority['priority']) => {
+        setData((prevData) => ({
+            ...prevData,
+            priority,
+            persistent_notifications: undefined, // Uncheck if checked already
+        }));
+    }, []);
+
+    const handleUpdateRequestedAck = useCallback((requested_ack: boolean) => {
+        setData((prevData) => ({...prevData, requested_ack}));
+    }, [data]);
+
+    const handleUpdatePersistentNotifications = useCallback((persistent_notifications: boolean) => {
+        setData((prevData) => ({...prevData, persistent_notifications}));
+    }, [data]);
+
+    const handleSubmit = useCallback(() => {
+        updatePostPriority(data);
+        close();
+    }, [data]);
+
+    const renderAcknowledgementOption = () => (
```

Comment:
for readability, consider moving this to it's own separate component, same with the others

Reply:
> for readability, consider moving this to it's own separate component

@enahum all these options just render one component `PickerOption` I don't see any benefit in readability since the result would be the same, still rendering one component.